scripts/Connector/Connector_JST/conn_jst_XA_vertical.py

```diff
@@ -0,0 +1,251 @@
+#sys.path.append(os.path.join(sys.path[0],"..","..","kicad_mod")) # load kicad_mod path
+
+
+# export PYTHONPATH="${PYTHONPATH}<path to kicad-footprint-generator directory>"
+sys.path.append(os.path.join(sys.path[0], "..", "..", ".."))  # load parent path of KicadModTree
+import argparse
+import yaml
+from helpers import *
+from KicadModTree import *
+
+sys.path.append(os.path.join(sys.path[0], "..", "..", "tools"))  # load parent path of tools
+from footprint_text_fields import addTextFields
+
+series = "XA"
+manufacturer = 'JST'
+orientation = 'V'
+number_of_rows = 1
+datasheet = 'http://www.jst-mfg.com/product/pdf/eng/eXA1.pdf'
+
+
+fab_pin1_marker_type = 1
+pin1_marker_offset = 0.3
+pin1_marker_linelen = 1.25
+
+drill_size = 0.95 #Datasheet: 0.9 +0.1/-0
+mh_drill = 1.25 #for boss
+pad_to_pad_clearance = 0.8
+pad_copper_y_solder_length = 0.5 #How much copper should be in y direction?
+min_annular_ring = 0.15
+
+
+
+pitch = 2.5
+
+# as of 2020 : available in 2- to 15-pin, and 20-pin variants.
+pincounts = [*range(2,15+1),20]
+
+variant_parameters = {
+    '1-A': {
+        'boss':True,
+        'pin_range':pincounts,
+        'descr_str':', with boss'
+        },
+    '1': {
+        'boss':False,
+        'pin_range':pincounts,
+        'descr_str':''
+        }
+}
+def generate_one_footprint(pincount, variant, configuration):
+    mpn = "B{pincount:02}B-XASK-{suff}".format(pincount=pincount,suff=variant)
+    boss = variant_parameters[variant]['boss']
+
+    A = (pincount - 1)*pitch
+    B = A + 5
+    x_min = (A-B)/2
+    y_max = 3.2
+    y_min = y_max - 6.4
+
+    silk_x_min = x_min - configuration['silk_fab_offset']
+    silk_y_min = y_min - configuration['silk_fab_offset']
+    silk_y_max = y_max + configuration['silk_fab_offset']
+
+
+    x_mid = A/2
+    x_max = (A+B)/2
+    silk_x_max = x_max + configuration['silk_fab_offset']
+
+    # Through-hole type shrouded header, Top entry type
+    orientation_str = configuration['orientation_options'][orientation]
+    footprint_name = configuration['fp_name_format_string'].format(man=manufacturer,
+        series=series,
+        mpn=mpn, num_rows=number_of_rows, pins_per_row=pincount, mounting_pad = "",
+        pitch=pitch, orientation=orientation_str)
+
+    kicad_mod = Footprint(footprint_name)
+    kicad_mod.setDescription("JST {:s} series connector, {:s} ({:s}), generated with kicad-footprint-generator".format(series, mpn, datasheet))
+    tags = configuration['keyword_fp_string'].format(series=series,
+        orientation=orientation_str, man=manufacturer,
+        entry=configuration['entry_direction'][orientation])
+    if boss:
+        tags += ' boss'
+    kicad_mod.setTags(tags)
+
+    ########################### Silkscreen ###########################
+    #Note, the locking recess dimensions vary with pincount, so this is approximate and mostly incorrect.
+    if pincount <= 3:
+        bump = 0.8
+        recess = 3.2
+    else:
+        bump = 1.6
+        recess = 5.4
```

Comment:
so i took a look at this. seems like the following would be what the parts look like:
```
bump = 0.8 if pincount == 2 else 1.6
if pincount > 3:
    recess = 5.4
else:
    recess = 3.2 if pincount == 2 else 4.8
```
starting with `pincount == 6` the parts have a second recess on each side.
i didn't look into how to create those, or their measurements

Reply:
I'm not a fan of that notation (with  "recess = 3.2 if ... " already in an else block), but if I understand right :
pincount==2 : (bump=0.8 , recess = 3.2)
pincount==3: (bump=1.6, recess=4.8)
pincount>=4: (bump=1.6, recess=5.4) ?

Reply:
sure, i'm not hung up on how to write it, just wanted to get the information across :)

Reply:
Perfect, I'll add the missing recess=4.8 case.

Reply:
and i just saw this was a question.
yes, you understood right

Reply:
Oops forgot to mention, I didn't look at the 2-recess type for the >= 6 pins. I vaguely recall looking at a few different 3D models, not finding an obvious logic in the recess layout & sizes, and quickly giving up !

Reply:
i'll take a look and see if i can figure it out easily.
i'll be back with some measurements if i manage

Reply:
ok so i spent some time with inkscape and screenshots of the connectors today..

it seems like all connectors with >=6 pins have the same following measurements:
![2020-04-06-105347](https://user-images.githubusercontent.com/22578460/78540826-01da5480-77f5-11ea-97fa-90f710a285fa.png)
where C is the already known recess (5.4mm)
A = 2.3 and
B = 2.1
i'm not 100% sure about these measurements, but they should be close enough for the silkscreen layer (and +- 0.1 mm wouldn't make that much of a difference anyway)

would you mind trying this out?
or do you want me to take a stab at it?

Reply:
well, getting the silkscreen to look "nearly perfect" is certainly nice, but TBH I don't feel like putting in much more effort for non-functional improvements. If you're happy to do it, perfect ! but as far as I'm concerned this could be merged as-is and I bet few people will complain that the silk is off by 1mm or +- 1bump/recess  : )

Reply:
yeah i get that feeling ;)
could you maybe push the updates you have made in the meantime?
maybe i'll find some time later this week to cut me teeth on this if nobody merges it first. which i'd be totally ok with!

Reply:
Perfect. just force-pushed a few minutes ago